src/support/hazard.c

```diff
@@ -328,6 +328,10 @@ __wt_hazard_check(WT_SESSION_IMPL *session,
 	WT_SESSION_IMPL *s;
 	uint32_t i, j, hazard_inuse, max, session_cnt, walk_cnt;
 
+	/* If a file can never be evicted, hazard pointers aren't required. */
```

Comment:
Is this safe? Do we need hazard pointers to manage when it's OK to do in-memory splits - which are part of in memory.

Reply:
That same check is already in `__wt_hazard_set()` and `__wt_hazard_clear()`, the hazard pointer set/clear functions, this just adds the check to the eviction path.

Reply:
I thought there was a check somewhere, but couldn't spot it in 2 minutes.